Library/Homebrew/cmd/update.sh

```diff
@@ -368,25 +368,29 @@ user account:
 EOS
   fi
 
-  # we may want to use a Homebrew curl
   if [[ -n "$HOMEBREW_FORCE_BREWED_CURL" &&
       ! -x "$HOMEBREW_PREFIX/opt/curl/bin/curl" ]]
   then
-    brew install curl
+    ohai "Installing Homebrew cURL"
+    if [[ -d "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" ]] && ! brew install curl
+    then
+      odie "Curl must be installed and in your PATH!"
+    fi
   fi
 
   if ! git --version &>/dev/null ||
      [[ -n "$HOMEBREW_FORCE_BREWED_GIT" &&
       ! -x "$HOMEBREW_PREFIX/opt/git/bin/git" ]]
   then
-    # we cannot install brewed git if homebrew/core is unavailable.
+    ohai "Installing Homebrew Git"
     [[ -d "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" ]] && brew install git
     unset GIT_EXECUTABLE
     if ! git --version &>/dev/null
     then
       odie "Git must be installed and in your PATH!"
     fi
```

Comment:
@MikeMcQuaid, how about this?
```suggestion
    if [[ -d "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" ]] && ! brew install git
    then
      odie "Git must be installed and in your PATH!"
    fi
```